lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -378,15 +390,24 @@ export abstract class ClientApplication {
      * object. If no authority set in application object, then default to common authority.
      * @param authorityString - authority from user configuration
      */
-    private async createAuthority(authorityString: string, azureRegionConfiguration?: AzureRegionConfiguration, requestCorrelationId?: string): Promise<Authority> {
+    private async createAuthority(authorityString: string, azureRegionConfiguration?: AzureRegionConfiguration, requestCorrelationId?: string, azureCloudInstance?: AzureCloudInstance): Promise<Authority> {
         this.logger.verbose("createAuthority called", requestCorrelationId);
+
+        // build authority string based on auth params - azureCloudInstance is prioritized if provided
+        let authorityAzureCloudInstance;
+        if (azureCloudInstance) {
+            authorityAzureCloudInstance = `${Authority.getAzureCloudInstanceUrl(azureCloudInstance)}/${Constants.DEFAULT_AUTHORITY_TENANT}/`;
```

Comment:
we're doing this in a couple different places, can we design this in a way that this only needs to be done once?

Reply:
Only two places: Configuration and Request which cannot be avoided. We can do a utils API may be. I optimized it for `Request` such that all `Request Types` can reuse the code.

Reply:
Addressed this.